src/main/java/duke/io/Parser.java

```diff
@@ -0,0 +1,187 @@
+     *
+     * @param tasks the list the parser stores tasks into.
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Parses input and handles user input accordingly.
+     *
+     * @param input command given by user input.
+     * @throws DukeException exception thrown due to invalid command or file writing error.
+     */
+    public String parse(String input) throws DukeException {
+        Command command;
+        String[] userInput = input.split(" ", 2);
+
+        if (!userInput[0].equals(userInput[0].toLowerCase())) {
+            throw new DukeException(
+                    String.format("Please input command with lowercase! Did you mean %s?", userInput[0].toLowerCase()));
+        }
+
+        try {
+            command = Command.valueOf(userInput[0].toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+        String message = "";
+        switch (command) {
+        case LIST:
+            message = tasks.print();
+            break;
+        case DONE:
+            message = setTaskDone(userInput);
+            break;
+        case TODO:
+            message = addTodo(userInput);
+            break;
+        case DEADLINE:
+            message = addDeadline(userInput);
+            break;
+        case EVENT:
+            message = addEvent(userInput);
+            break;
+        case DELETE:
+            message = deleteTask(userInput);
+            break;
+        case FIND:
+            message = findTask(userInput);
+            break;
+        case BYE:
+            message = Ui.sayBye();
+            break;
+        default:
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+        return message;
+    }
+
+    /**
+     * Adds the input to the list of tasks and prints out the input.
+     *
+     * @param input to be added and printed.
+     * @return string to inform user of successful command.
+     */
+    private String addTask(Task input) {
+        tasks.add(input);
+        return Ui.print(String.format("Got it. I've added this task:\n  %s\nNow you have %d %s in the list.",
+                input, tasks.size(), tasks.size() == 1 ? "task" : "tasks"));
+    }
+
+    /**
+     * Adds a todo task to the list of tasks.
+     *
+     * @param userInput given by user.
+     * @return string to inform user of successful command.
+     */
+    private String addTodo(String... userInput) throws DukeException {
+        try {
+            return addTask(new Todo(userInput[1]));
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Todo description cannot be empty");
+        }
+    }
+
+    /**
+     * Adds a deadline task to the list of tasks.
+     *
+     * @param userInput given by user.
+     * @return string to inform user of successful command.
+     */
```

Comment:
Perhaps include a `@throws` javadoc here? It might apply to similar javadocs in this file as well.